Source/objects.cpp

```diff
@@ -4822,16 +4823,19 @@ void OperateTrap(Object &trap)
 	}
 
 	trap._oVar4 = 1;
-	Point target = trigger.position;
-	for (int y = target.y - 1; y <= trigger.position.y + 1; y++) {
-		for (int x = trigger.position.x - 1; x <= trigger.position.x + 1; x++) {
-			if (dPlayer[x][y] != 0) {
-				target.x = x;
-				target.y = y;
-			}
-		}
-	}
+
 	if (!deltaload) {
+		// default to firing at the trigger object
+		Point target = trigger.position;
+
+		PointsInRectangleRange searchArea { Rectangle { target, 1 } };
+		// look for a player near the trigger (using a reverse search to match vanilla behaviour)
+		auto foundPosition = std::find_if(searchArea.crbegin(), searchArea.crend(), [](Point testPosition) { return InDungeonBounds(testPosition) && dPlayer[testPosition.x][testPosition.y] != 0; });

```

Comment:
Would it make sens to have `InDungeonBounds()` be part of `PointsInRectangleRange`?

Reply:
I don't think that would be appropriate for this iterator, it's more concerned with providing a way to access all discrete Points that are contained within a Rectangle. If we wanted to restrict creating rectangles to only in bounds areas then that'd be best enforced in the Rectangle class (but I wouldn't want to do that either). For this case, I'd rather use a wrapper `IsPlayerAtPosition` similar to the `IsObjectAtPosition` helper that is proposed in #3324